src/CommonProviderImplementation/Helpers.fs

```diff
@@ -332,17 +334,18 @@ module internal ProviderHelpers =
         let defaultResolutionFolder = cfg.ResolutionFolder
         
         let parse extension (value:string) = using (logTime "Parsing" sampleOrSampleUri) <| fun _ ->
-            if sampleIsList then
-                parseList extension value
-            else
-                [| parseSingle extension value |]
+            lazy(
+                if sampleIsList then
+                    parseList extension value
+                else
+                    [| parseSingle extension value |])
```

Comment:
We should lazy for values which are initialized but might not be needed on all the forward the code paths, but I don't think it makes sense to use lazy on the output of functions